syscall_sdl.c

```diff
@@ -43,10 +111,64 @@ static bool check_sdl(struct riscv_t *rv, uint32_t width, uint32_t height)
             rv_halt(rv);
             return false;
         case SDL_KEYDOWN:
-            if (event.key.keysym.sym == SDLK_ESCAPE) {
-                rv_halt(rv);
-                return false;
+            if (event.key.keysym.sym == SDLK_ESCAPE &&
+                SDL_GetRelativeMouseMode() == SDL_TRUE) {
+                SDL_SetRelativeMouseMode(SDL_FALSE);
+                break;
             }
+            /* fall through */
+        case SDL_KEYUP: {
+            if (event.key.repeat)
+                break;
+            event_t new_event = {
+                .type = KEY_EVENT,
+                .key_event =
+                    {
+                        .keycode = event.key.keysym.sym,
+                        .state = (bool) (event.key.state == SDL_PRESSED),
+                    },
+            };
+            event_push(new_event);
+            break;
+        }
+        case SDL_MOUSEMOTION: {
+            event_t new_event = {
+                .type = MOUSE_MOTION_EVENT,
+                .mouse =
```

Comment:
Use `memcpy` to initiate `motion` member, resulting in shorter code and a bit faster execution.